dotnet/src/Plugins/Plugins.Core/CodeInterpreter/SessionsPythonPlugin.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Plugins.Core.CodeInterpreter;
+
+/// <summary>
+/// A plugin for running Python code in an Azure Container Apps dynamic sessions code interpreter.
+/// </summary>
+public class SessionsPythonPlugin
+{
+    private static readonly string s_assemblyVersion = typeof(Kernel).Assembly.GetName().Version!.ToString();
+    private readonly Uri _poolManagementEndpoint;
+    private readonly SessionPythonSettings _settings;
+    private readonly Func<Task<string>> _authTokenProvider;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly ILogger _logger;
+    private string? _authToken;
+
+    /// <summary>
+    /// Initializes a new instance of the SessionsPythonTool class.
+    /// </summary>
+    /// <param name="settings">The settings for the Python tool plugin. </param>
+    /// <param name="authTokenProvider"> The function to provide the auth token. </param>
+    /// <param name="httpClientFactory">The HTTP client factory. </param>
+    /// <param name="loggerFactory">The logger factory. </param>
+    public SessionsPythonPlugin(SessionPythonSettings settings, Func<Task<string>> authTokenProvider, IHttpClientFactory httpClientFactory, ILoggerFactory? loggerFactory = null)
+    {
+        Verify.NotNull(settings, nameof(settings));
+        Verify.NotNull(authTokenProvider, nameof(authTokenProvider));
+        Verify.NotNull(httpClientFactory, nameof(httpClientFactory));
+        Verify.NotNull(settings.Endpoint, nameof(settings.Endpoint));
+
+        this._settings = settings;
+
+        // Ensure the endpoint won't change by reference 
```

Comment:
FYI, The Endpoint prop is init only, and Uri is immutable, so changing by reference shouldn't be possible in this case.

Reply:
Endpoint shouldn't change, moved those to the constructor.